goreq.go

```diff
@@ -309,7 +315,7 @@ func (r Request) Do() (*Response, error) {
 
 	timeout := false
 	var timer *time.Timer
-	if r.Timeout > 0 {
+	if transport, ok := transport.(transportRequestCanceler); ok && r.Timeout > 0 {
```

Comment:
I wanted to be as general as possible - if theres a Transport that supports `CancelRequest` that isn't the `http.Transport` it would be nice if request timeouts still worked.

That said the http2 Transport doesn't support this, and I'm not aware of any other `RoundTripper`s in the wild.